nav2_collision_monitor/src/circle.cpp

```diff
@@ -58,14 +58,20 @@ void Circle::getPolygon(std::vector<Point> & poly) const
   }
 }
 
-int Circle::getPointsInside(const std::vector<Point> & points) const
+int Circle::getPointsInside(
+  const std::vector<Point> & points,
+  std::vector<std::string> & source_names) const
 {
   int num = 0;
-  for (Point point : points) {
+  std::unordered_set<std::string> unordered_source_names;
+  for (const Point & point : points) {
     if (point.x * point.x + point.y * point.y < radius_squared_) {
       num++;
+      unordered_source_names.insert(point.source_name);
     }
   }
+  source_names.assign(unordered_source_names.begin(), unordered_source_names.end());
```

Comment:
Unless the source names stuff is needed for the watchdog, lets remove it. This PR is starting to get a little too complicated to review with this change in here if its a new feature.

Also, this seems a little convoluted. If we were to add this, I'd think changing points to being a vector of sources, whereas each source is a vector of points + a source string name. That way we iterate through a vector of a vector of points, where each nested vector as a single source name attached to it so we don't have these massively long source_name vectors of repeating values